nixos/modules/services/x11/desktop-managers/cinnamon.nix

```diff
@@ -0,0 +1,155 @@
+        (defaultPackages ++ cfg.extraGSettingsOverridePackages)}
+
+     cp -f ${pkgs.gnome3.gnome-shell}/share/gsettings-schemas/*/glib-2.0/schemas/*.gschema.override $out/share/gsettings-schemas/nixos-gsettings-overrides/glib-2.0/schemas
+
+     chmod -R a+w $out/share/gsettings-schemas/nixos-gsettings-overrides
+     cat - > $out/share/gsettings-schemas/nixos-gsettings-overrides/glib-2.0/schemas/nixos-defaults.gschema.override <<- EOF
+       [org.gnome.desktop.background]
+       picture-uri='file://${pkgs.nixos-artwork.wallpapers.simple-dark-gray}/share/artwork/gnome/nix-wallpaper-simple-dark-gray.png'
+
+       [org.gnome.desktop.screensaver]
+       picture-uri='file://${pkgs.nixos-artwork.wallpapers.simple-dark-gray-bottom}/share/artwork/gnome/nix-wallpaper-simple-dark-gray_bottom.png'
+
+       [org.gnome.shell]
+       favorite-apps=[ 'org.gnome.Epiphany.desktop', 'org.gnome.Geary.desktop', 'org.gnome.Music.desktop', 'org.gnome.Photos.desktop', 'org.gnome.Nautilus.desktop', 'org.gnome.Software.desktop' ]
+
+       ${cfg.extraGSettingsOverrides}
+     EOF
+
+     ${pkgs.glib.dev}/bin/glib-compile-schemas $out/share/gsettings-schemas/nixos-gsettings-overrides/glib-2.0/schemas/
+    '';
+in
+
+{
+
+  options = {
+    services.xserver.desktopManager.cinnamon = {
+      enable = mkOption {
+        type = types.bool;
+        default = false;
+        description = "Enable the Cinnamon desktop environment.";
+      };
+
+      extraGSettingsOverrides = mkOption {
+        default = "";
+        type = types.lines;
+        description = "Additional gsettings overrides.";
+      };
+
+      extraGSettingsOverridePackages = mkOption {
+        default = [];
+        type = types.listOf types.path;
+        description = "List of packages for which gsettings are overridden.";
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+    # Enable colord server
+    services.colord.enable = mkDefault true;
+
+    # Enable dconf
+    programs.dconf.enable = mkDefault true;
+
+    # Enable org.a11y.Bus
+    services.gnome3.at-spi2-core.enable = true;
+
+    # Fix lockscreen
+    security.pam.services = {
+      cinnamon-screensaver = {};
+    };
+
+    # Power managment support
+    services.upower.enable = mkDefault true;
+
+    # Network manager
+    networking.networkmanager.enable = mkDefault true;
+
+    # Override GSettings schemas
+    environment.sessionVariables.NIX_GSETTINGS_OVERRIDES_DIR = "${nixos-gsettings-desktop-schemas}/share/gsettings-schemas/nixos-gsettings-overrides/glib-2.0/schemas";
+
+    environment.systemPackages = with pkgs.cinnamon // pkgs; [
+      # common-files
+      cinnamon-common
+      cinnamon-session
+      cinnamon-desktop
+
+      # utils needed by some scripts
+      pkgs.killall
+
+      # session requirements
+      cinnamon-screensaver
+      # nemo-autostart: provided by nemo
+      pkgs.gnome3.networkmanagerapplet
```

Comment:
is it that we need the service running `programs.nm-applet.enable`. We're preferring to use this over the xdg-autostart in the package.

Reply:
It was saying something about needing that or smth. Don't remember exactly but if nm-applet gets the job done than that works too